frontend/src/app/team-management/search-team-management/search-team-management.component.ts

```diff
@@ -0,0 +1,147 @@
+import { Component } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { UserService } from '../../services/user.service';
+import { TeamService } from '../../services/team.service';
+import { Team } from '../../shared/types/model/Team';
+import { User } from '../../shared/types/model/User';
+import { BehaviorSubject, combineLatest, debounceTime, distinctUntilChanged, map } from 'rxjs';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { getRouteToTeam, getRouteToUserDetails } from '../../shared/routeUtils';
+import { Router } from '@angular/router';
+
+export interface FilteredUser extends User {
+  displayValue: string;
+  htmlValue: string;
+}
+
+export interface FilteredTeam extends Team {
+  displayValue: string;
+  htmlValue: string;
+}
+
+@Component({
+  selector: 'app-search-team-management',
+  templateUrl: './search-team-management.component.html',
+  styleUrl: './search-team-management.component.scss',
+})
+export class SearchTeamManagementComponent {
+  static MAX_SUGGESTIONS = 3;
+  search = new FormControl('');
+
+  filteredUsers$ = new BehaviorSubject<FilteredUser[]>([]);
+  filteredTeams$ = new BehaviorSubject<FilteredTeam[]>([]);
+  searchValue$ = new BehaviorSubject<string>('');
+
+  private teams: Team[] = [];
+  private users: User[] = [];
+
+  constructor(
+    private readonly userService: UserService,
+    private readonly teamService: TeamService,
+    private readonly router: Router,
+  ) {
+    combineLatest([teamService.getAllTeams(), userService.getUsers()])
+      .pipe(takeUntilDestroyed())
+      .subscribe(([teams, users]) => {
+        this.updateTeamsAndUsers(teams, users);
+        this.applyFilter(this.searchValue$.getValue());
+      });
+
+    this.search.valueChanges
+      .pipe(
+        takeUntilDestroyed(),
+        debounceTime(200),
+        map((v) => (v ? v.trim() : '')),
+        distinctUntilChanged(),
+      )
+      .subscribe((searchValue) => {
+        this.searchValue$.next(searchValue);
+      });
+
+    this.searchValue$.pipe(takeUntilDestroyed()).subscribe(() => {
+      this.applyFilter(this.searchValue$.getValue());
+    });
+  }
+
+  selectUser(user: User) {
+    this.search.setValue('');
+    this.router.navigateByUrl(getRouteToUserDetails(user.id)).then();
+  }
+
+  selectTeam(team: Team) {
+    this.search.setValue('');
+    this.router.navigateByUrl(getRouteToTeam(team.id)).then();
+  }
+
+  private applyFilter(filterValue: string): void {
+    if (filterValue.length == 0) {
```

Comment:
ist == hier Absicht?

Reply:
Fixed in https://github.com/puzzle/okr/pull/758/commits/3f193c6ce59a8469467c0ccdbb1e025a56b5e949

Reply:
Fixed in https://github.com/puzzle/okr/pull/758/commits/3f193c6ce59a8469467c0ccdbb1e025a56b5e949